torch/_inductor/codegen/cpp.py

```diff
@@ -2986,11 +2986,16 @@ def store_reduction(self, name, index, value):
         index = self.rename_indexing(index)
         var = self.args.output(name)
         out_dtype = V.graph.get_dtype(name)
-        dtype = (
-            (out_dtype if out_dtype == torch.double else torch.float)
-            if out_dtype.is_floating_point
-            else torch.int64
-        )
+        if out_dtype.is_floating_point:
+            if out_dtype == torch.double:
+                dtype = out_dtype
+            else:
+                dtype = torch.float
+        else:
+            if out_dtype == torch.int64:
+                dtype = out_dtype
+            else:
+                dtype = torch.int
```

Comment:
Would there be unnecessary conversion when `out_dtype` is int8? Would the following work?
```suggestion
        dtype = value.dtype if isinstance(value, CppCSEVariable) else out_dtype
```

Reply:
@jgong5 
1. Yes, based on the changes mentioned above, there is an unnecessary conversion when the output dtype is int8.
2. ```dtype = value.dtype if isinstance(value, CppCSEVariable) else out_dtype```  cannot work because it would cause a core dump. I believe the reason is that the value is always a string, not a CppCSEVariable. This leads to the conversion to torch.float32 being omitted when out_dtype is torch.float16 or torch.bfloat16, unlike the previous implementation.
3. So I've kept the float logic unchanged and modified the integer logic to dtype = out_dtype. Please take a look at the latest PR.